source/lib.ts

```diff
@@ -284,86 +305,67 @@ const handleAsyncErrors =
 	const config = parseOptions(passedOptions ?? {})
 	const options = getOptionsFromConfig(config)
 
-	// Call the `init` method on the store, if it exists
+	// Call the `init` method on the store, if it exists.
 	if (typeof config.store.init === 'function') config.store.init(options)
 
-	// Then return the actual middleware
+	// Then return the actual middleware.
 	const middleware = handleAsyncErrors(
 		async (request: Request, response: Response, next: NextFunction) => {
-			// First check if we should skip the request
+			// First check if we should skip the request.
 			const skip = await config.skip(request, response)
 			if (skip) {
 				next()
 				return
 			}
 
-			// Create an augmented request
+			// Create an augmented request.
 			const augmentedRequest = request as AugmentedRequest
 
-			// Get a unique key for the client
+			// Get a unique key for the client, then increment the client's hit counter
+			// by one, and make sure it's only by one.
 			const key = await config.keyGenerator(request, response)
-			// Increment the client's hit counter by one, and make sure it's only by one
 			const { totalHits, resetTime } = await config.store.increment(key)
 			config.validations.singleCount(request, config.store, key)
 
-			// Get the quota (max number of hits) for each client
+			// Get the quota (max number of hits) for each client.
 			const retrieveQuota =
 				typeof config.max === 'function'
 					? config.max(request, response)
 					: config.max
 			const maxHits = await retrieveQuota
-			config.validations.max(maxHits)
+			config.validations.max(maxHits) // Warn about the change in `max: 0` behaviour.
 
 			// Set the rate limit information on the augmented request object
-			augmentedRequest[config.requestPropertyName] = {
+			const info: RateLimitInfo = {
 				limit: maxHits,
 				current: totalHits,
 				remaining: Math.max(maxHits - totalHits, 0),
 				resetTime,
 			}
+			augmentedRequest[config.requestPropertyName] = info
 
-			// Set the X-RateLimit headers on the response object if enabled
+			// Set the `X-RateLimit` headers on the response object if enabled.
 			if (config.legacyHeaders && !response.headersSent) {
-				response.setHeader('X-RateLimit-Limit', maxHits)
-				response.setHeader(
-					'X-RateLimit-Remaining',
-					augmentedRequest[config.requestPropertyName].remaining,
-				)
-
-				// If we have a resetTime, also provide the current date to help avoid
-				// issues with incorrect clocks.
-				if (resetTime instanceof Date) {
-					response.setHeader('Date', new Date().toUTCString())
-					response.setHeader(
-						'X-RateLimit-Reset',
-						Math.ceil(resetTime.getTime() / 1000),
-					)
-				}
+				setLegacyHeaders(response, info)
 			}
 
-			// Set the standardized RateLimit headers on the response object
-			// if enabled.
+			// Set the standardized `RateLimit` and `RateLimit-Policy` headers on the
+			// response object if enabled.
```

Comment:
This is true for draft 7, but this comment covers drafts 6 & 7 - perhaps we should move or reword it

```suggestion
			// Set the standardized `RateLimit*` headers on the
			// response object if enabled.
```